apps/DesktopStreamer/Stream.cpp

```diff
@@ -0,0 +1,230 @@
+/*   2. Redistributions in binary form must reproduce the above      */
+/*      copyright notice, this list of conditions and the following  */
+/*      disclaimer in the documentation and/or other materials       */
+/*      provided with the distribution.                              */
+/*                                                                   */
+/*    THIS  SOFTWARE IS PROVIDED  BY THE  UNIVERSITY OF  TEXAS AT    */
+/*    AUSTIN  ``AS IS''  AND ANY  EXPRESS OR  IMPLIED WARRANTIES,    */
+/*    INCLUDING, BUT  NOT LIMITED  TO, THE IMPLIED  WARRANTIES OF    */
+/*    MERCHANTABILITY  AND FITNESS FOR  A PARTICULAR  PURPOSE ARE    */
+/*    DISCLAIMED.  IN  NO EVENT SHALL THE UNIVERSITY  OF TEXAS AT    */
+/*    AUSTIN OR CONTRIBUTORS BE  LIABLE FOR ANY DIRECT, INDIRECT,    */
+/*    INCIDENTAL,  SPECIAL, EXEMPLARY,  OR  CONSEQUENTIAL DAMAGES    */
+/*    (INCLUDING, BUT  NOT LIMITED TO,  PROCUREMENT OF SUBSTITUTE    */
+/*    GOODS  OR  SERVICES; LOSS  OF  USE,  DATA,  OR PROFITS;  OR    */
+/*    BUSINESS INTERRUPTION) HOWEVER CAUSED  AND ON ANY THEORY OF    */
+/*    LIABILITY, WHETHER  IN CONTRACT, STRICT  LIABILITY, OR TORT    */
+/*    (INCLUDING NEGLIGENCE OR OTHERWISE)  ARISING IN ANY WAY OUT    */
+/*    OF  THE  USE OF  THIS  SOFTWARE,  EVEN  IF ADVISED  OF  THE    */
+/*    POSSIBILITY OF SUCH DAMAGE.                                    */
+/*                                                                   */
+/* The views and conclusions contained in the software and           */
+/* documentation are those of the authors and should not be          */
+/* interpreted as representing official policies, either expressed   */
+/* or implied, of The Ecole Polytechnique Federal de Lausanne.       */
+/*********************************************************************/
+
+#include "Stream.h"
+#include "MainWindow.h"
+
+#ifdef __APPLE__
+#  ifdef DEFLECT_USE_QT5MACEXTRAS
+#    include "DesktopWindowsModel.h"
+#  endif
+#  if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_9
+#    include <CoreGraphics/CoreGraphics.h>
+#  else
+#    include <ApplicationServices/ApplicationServices.h>
+#  endif
+#endif
+#include <QPainter>
+#include <QScreen>
+
+#define CURSOR_IMAGE_FILE     ":/cursor.png"
+#define CURSOR_IMAGE_SIZE     20
+
+Stream::Stream( const MainWindow& parent, const QPersistentModelIndex window,
+                const std::string& name, const std::string& host )
+    : deflect::Stream( name, host )
+    , _parent( parent )
+    , _window( window )
+    , _cursor( QImage( CURSOR_IMAGE_FILE ).scaled(
+                   CURSOR_IMAGE_SIZE * parent.devicePixelRatio(),
+                   CURSOR_IMAGE_SIZE * parent.devicePixelRatio(),
+                   Qt::KeepAspectRatio ))
+{}
+
+Stream::~Stream()
+{}
+
+bool Stream::processEvents( const bool interact )
+{
+    while( isRegisteredForEvents() && hasEvent( ))
+    {
+        deflect::Event event = getEvent();
```

Comment:
right. Events are always processed in non-retina space := no conversion needed. Only the cursor rendering on the retina pixmap needs this. KISS